diff -up gnome-settings-daemon-3.5.4/configure.ac.ppc-no-wacom gnome-settings-daemon-3.5.4/configure.ac
--- gnome-settings-daemon-3.5.4/configure.ac.ppc-no-wacom	2012-07-17 18:42:41.000000000 +0200
+++ gnome-settings-daemon-3.5.4/configure.ac	2012-07-17 18:43:39.000000000 +0200
@@ -248,12 +248,12 @@ fi
 AC_SUBST(GEOCLUE_DBUS_INTERFACE_XML)
 
 dnl ---------------------------------------------------------------------------
-dnl - wacom (disabled for s390/s390x and non Linux platforms)
+dnl - wacom (disabled for s390/s390x, ppc/ppc64 and non Linux platforms)
 dnl ---------------------------------------------------------------------------
 
 case $host_os in
   linux*)
-    if test "$host_cpu" = s390 -o "$host_cpu" = s390x; then
+    if test "$host_cpu" = s390 -o "$host_cpu" = s390x -o "$host_cpu" = powerpc -o "$host_cpu" = ppc -o "$host_cpu" = powerpc64 -o "$host_cpu" = ppc64 ; then
       have_wacom=no
     else
       if test x$enable_gudev != xno; then
