diff -up epiphany-3.11.92/src/bookmarks/ephy-bookmarks.c.default-bookmarks epiphany-3.11.92/src/bookmarks/ephy-bookmarks.c
--- epiphany-3.11.92/src/bookmarks/ephy-bookmarks.c.default-bookmarks	2014-02-04 10:55:03.000000000 -0500
+++ epiphany-3.11.92/src/bookmarks/ephy-bookmarks.c	2014-03-20 12:50:04.377875262 -0400
@@ -116,6 +116,12 @@ ephy_bookmarks_init_defaults (EphyBookma
 {
 	int i;
 
+	if (g_file_test ("/usr/share/bookmarks/default-bookmarks.html", G_FILE_TEST_EXISTS))
+	{
+		ephy_bookmarks_import_mozilla (eb, "/usr/share/bookmarks/default-bookmarks.html");
+		return;
+	}
+
 	for (i = 0; i < G_N_ELEMENTS (default_topics); i++)
 	{
 		ephy_bookmarks_add_keyword (eb, _(default_topics[i]));
