diff --git a/src/ephy-notebook.c b/src/ephy-notebook.c
index 2643522..6a9c1b1 100644
--- a/src/ephy-notebook.c
+++ b/src/ephy-notebook.c
@@ -37,7 +37,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
-#define TAB_WIDTH_N_CHARS 15
+#define TAB_WIDTH_N_CHARS 5
 
 #define AFTER_ALL_TABS -1
 #define NOT_IN_APP_WINDOWS -2
@@ -561,12 +561,12 @@ build_tab_label (EphyNotebook *nb, EphyEmbed *embed)
 	GtkWidget *box;
 	EphyWebView *view;
 
-	box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 4);
+	box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
 	gtk_widget_show (box);
 
 	/* set hbox spacing and label padding (see below) so that there's an
 	 * equal amount of space around the label */
-	hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 6);
+	hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 2);
 	gtk_widget_show (hbox);
 	gtk_widget_set_halign (hbox, GTK_ALIGN_CENTER);
 	gtk_box_pack_start (GTK_BOX (box), hbox, TRUE, TRUE, 0);
