diff --git a/geiser-guile.el b/geiser-guile.el
index db42945..dc691f5 100644
--- a/geiser-guile.el
+++ b/geiser-guile.el
@@ -48,7 +48,7 @@
 (geiser-custom--defcustom geiser-guile-binary
     (cond ((eq system-type 'windows-nt) "guile.exe")
           ((eq system-type 'darwin) "guile")
-          (t "guile"))
+          (t "guile3.0"))
   "Name to use to call the Guile executable when starting a REPL."
   :type '(choice string (repeat string)))
 
