diff --git a/elisp/geiser-guile.el b/elisp/geiser-guile.el
index 522c4eb..75e33c7 100644
--- a/elisp/geiser-guile.el
+++ b/elisp/geiser-guile.el
@@ -37,7 +37,7 @@
 (geiser-custom--defcustom geiser-guile-binary
     (cond ((eq system-type 'windows-nt) "guile.exe")
           ((eq system-type 'darwin) "guile")
-          (t "guile"))
+          (t "guile2.2"))
   "Name to use to call the Guile executable when starting a REPL."
   :type '(choice string (repeat string))
   :group 'geiser-guile)
