diff -up chromium-98.0.4758.102/base/third_party/symbolize/symbolize.h.missing-utility-for-std-exchange chromium-98.0.4758.102/base/third_party/symbolize/symbolize.h
--- chromium-98.0.4758.102/base/third_party/symbolize/symbolize.h.missing-utility-for-std-exchange	2022-02-25 22:30:02.833745309 +0000
+++ chromium-98.0.4758.102/base/third_party/symbolize/symbolize.h	2022-02-25 22:30:02.832745344 +0000
@@ -58,6 +58,8 @@
 #include "config.h"
 #include "glog/logging.h"
 
+#include <utility>
+
 #ifdef HAVE_SYMBOLIZE
 
 #include <algorithm>
