Index: sandbox/linux/BUILD.gn
diff --git a/sandbox/linux/BUILD.gn b/sandbox/linux/BUILD.gn
index 3070b7347bcc115bb1fc145d2162d9db366a6618..1e6d7a1c813296ff9e7d35ffb60340407a54cec7 100644
--- a/sandbox/linux/BUILD.gn
+++ b/sandbox/linux/BUILD.gn
@@ -41,10 +41,7 @@ group("sandbox") {
     public_deps += [ ":suid_sandbox_client" ]
   }
   if (use_seccomp_bpf || is_nacl_nonsfi) {
-    public_deps += [
-      ":seccomp_bpf",
-      ":seccomp_bpf_helpers",
-    ]
+    public_deps += [ ":seccomp_bpf" ]
   }
 }
 
@@ -221,6 +218,14 @@ component("seccomp_bpf") {
     "bpf_dsl/syscall_set.cc",
     "bpf_dsl/syscall_set.h",
     "bpf_dsl/trap_registry.h",
+    "seccomp-bpf-helpers/baseline_policy.cc",
+    "seccomp-bpf-helpers/baseline_policy.h",
+    "seccomp-bpf-helpers/sigsys_handlers.cc",
+    "seccomp-bpf-helpers/sigsys_handlers.h",
+    "seccomp-bpf-helpers/syscall_parameters_restrictions.cc",
+    "seccomp-bpf-helpers/syscall_parameters_restrictions.h",
+    "seccomp-bpf-helpers/syscall_sets.cc",
+    "seccomp-bpf-helpers/syscall_sets.h",
     "seccomp-bpf/die.cc",
     "seccomp-bpf/die.h",
     "seccomp-bpf/sandbox_bpf.cc",
@@ -250,31 +255,6 @@ component("seccomp_bpf") {
       "bpf_dsl/linux_syscall_ranges.h",
       "bpf_dsl/seccomp_macros.h",
       "bpf_dsl/trap_registry.h",
-    ]
-  }
-}
-
-component("seccomp_bpf_helpers") {
-  sources = [
-    "seccomp-bpf-helpers/baseline_policy.cc",
-    "seccomp-bpf-helpers/baseline_policy.h",
-    "seccomp-bpf-helpers/sigsys_handlers.cc",
-    "seccomp-bpf-helpers/sigsys_handlers.h",
-    "seccomp-bpf-helpers/syscall_parameters_restrictions.cc",
-    "seccomp-bpf-helpers/syscall_parameters_restrictions.h",
-    "seccomp-bpf-helpers/syscall_sets.cc",
-    "seccomp-bpf-helpers/syscall_sets.h",
-  ]
-  defines = [ "SANDBOX_IMPLEMENTATION" ]
-
-  deps = [
-    ":sandbox_services",
-    ":seccomp_bpf",
-    "//base",
-  ]
-
-  if (is_nacl_nonsfi) {
-    sources -= [
       "seccomp-bpf-helpers/baseline_policy.cc",
       "seccomp-bpf-helpers/baseline_policy.h",
       "seccomp-bpf-helpers/syscall_sets.cc",
Index: sandbox/linux/sandbox_linux.gypi
diff --git a/sandbox/linux/sandbox_linux.gypi b/sandbox/linux/sandbox_linux.gypi
index c19bdb1c5f67b1fcbebc0b772aae0d07b0ca0eb6..039e9b405174b708ad4fc36049f0f97c6b1ddcd3 100644
--- a/sandbox/linux/sandbox_linux.gypi
+++ b/sandbox/linux/sandbox_linux.gypi
@@ -54,7 +54,6 @@
         [ 'use_seccomp_bpf==1', {
           'dependencies': [
             'seccomp_bpf',
-            'seccomp_bpf_helpers',
           ],
         }],
       ],
@@ -141,6 +140,14 @@
         'bpf_dsl/syscall_set.cc',
         'bpf_dsl/syscall_set.h',
         'bpf_dsl/trap_registry.h',
+        'seccomp-bpf-helpers/baseline_policy.cc',
+        'seccomp-bpf-helpers/baseline_policy.h',
+        'seccomp-bpf-helpers/sigsys_handlers.cc',
+        'seccomp-bpf-helpers/sigsys_handlers.h',
+        'seccomp-bpf-helpers/syscall_parameters_restrictions.cc',
+        'seccomp-bpf-helpers/syscall_parameters_restrictions.h',
+        'seccomp-bpf-helpers/syscall_sets.cc',
+        'seccomp-bpf-helpers/syscall_sets.h',
         'seccomp-bpf/die.cc',
         'seccomp-bpf/die.h',
         'seccomp-bpf/sandbox_bpf.cc',
@@ -168,31 +175,6 @@
       ],
     },
     {
-      'target_name': 'seccomp_bpf_helpers',
-      'type': '<(component)',
-      'sources': [
-        'seccomp-bpf-helpers/baseline_policy.cc',
-        'seccomp-bpf-helpers/baseline_policy.h',
-        'seccomp-bpf-helpers/sigsys_handlers.cc',
-        'seccomp-bpf-helpers/sigsys_handlers.h',
-        'seccomp-bpf-helpers/syscall_parameters_restrictions.cc',
-        'seccomp-bpf-helpers/syscall_parameters_restrictions.h',
-        'seccomp-bpf-helpers/syscall_sets.cc',
-        'seccomp-bpf-helpers/syscall_sets.h',
-      ],
-      'dependencies': [
-        '../base/base.gyp:base',
-        'sandbox_services',
-        'seccomp_bpf',
-      ],
-      'defines': [
-        'SANDBOX_IMPLEMENTATION',
-      ],
-      'include_dirs': [
-        '../..',
-      ],
-    },
-    {
       # The setuid sandbox, for Linux
       'target_name': 'chrome_sandbox',
       'type': 'executable',
Index: services/shell/runner/host/BUILD.gn
diff --git a/services/shell/runner/host/BUILD.gn b/services/shell/runner/host/BUILD.gn
index 813067dece21a83f1422b244939e22e35f64edf8..3b01c0ee162001f827fe1d7404e25ff8bd10aa68 100644
--- a/services/shell/runner/host/BUILD.gn
+++ b/services/shell/runner/host/BUILD.gn
@@ -61,7 +61,6 @@ source_set("child_process_base") {
       "//sandbox/linux:sandbox",
       "//sandbox/linux:sandbox_services",
       "//sandbox/linux:seccomp_bpf",
-      "//sandbox/linux:seccomp_bpf_helpers",
     ]
   }
 
Index: services/shell/shell.gyp
diff --git a/services/shell/shell.gyp b/services/shell/shell.gyp
index eff0f628495388f034afd832a7bc8b8ffe8d9144..70349e2590d4e2a5d82198a79099c9eaf069926b 100644
--- a/services/shell/shell.gyp
+++ b/services/shell/shell.gyp
@@ -146,7 +146,6 @@
           '<(DEPTH)/sandbox/sandbox.gyp:sandbox',
           '<(DEPTH)/sandbox/sandbox.gyp:sandbox_services',
           '<(DEPTH)/sandbox/sandbox.gyp:seccomp_bpf',
-          '<(DEPTH)/sandbox/sandbox.gyp:seccomp_bpf_helpers',
         ],
       }],
       ['OS=="mac"', {
