--- chromium-60.0.3112.78/v8/src/objects-body-descriptors.h.gcc7	2017-07-26 03:06:36.000000000 +0800
+++ chromium-60.0.3112.78/v8/src/objects-body-descriptors.h	2017-07-27 03:01:10.330732724 +0800
@@ -99,7 +99,7 @@
 
   template <typename StaticVisitor>
   static inline void IterateBody(HeapObject* obj, int object_size) {
-    IterateBody(obj);
+    IterateBody<StaticVisitor>(obj);
   }
 
   static inline int SizeOf(Map* map, HeapObject* object) { return kSize; }
