diff --git a/ui/gfx/x/gen_xproto.py b/ui/gfx/x/gen_xproto.py
index 4a34901012d9..509f22c1f4e2 100644
--- a/ui/gfx/x/gen_xproto.py
+++ b/ui/gfx/x/gen_xproto.py
@@ -168,6 +168,7 @@ from __future__ import print_function
 
 import argparse
 import collections
+import functools
 import os
 import re
 import sys
@@ -919,11 +920,13 @@ class GenXproto:
                 return 3
             return 2
 
-        def cmp((_1, item1), (_2, item2)):
+        def cmp(arg1, arg2):
+            (_1, item1) = arg1
+            (_2, item2) = arg2
             return type_order_priority(item1) - type_order_priority(item2)
 
         # sort() is guaranteed to be stable.
-        self.module.all.sort(cmp=cmp)
+        self.module.all.sort(key=functools.cmp_to_key(cmp))
 
     def gen_header(self):
         self.file = self.args.headerfile
