--- chromium-71.0.3578.80/tools/gn/bootstrap/bootstrap.py.orig	2018-12-06 01:06:19.815698655 +0800
+++ chromium-71.0.3578.80/tools/gn/bootstrap/bootstrap.py	2018-12-06 01:07:32.262069775 +0800
@@ -94,6 +94,7 @@
       gn_gen_args += ' is_debug=false'
     subprocess.check_call([
         gn_path, 'gen', out_dir,
+        '--script-executable=/usr/bin/python2',
         '--args=%s' % gn_gen_args, "--root=" + SRC_ROOT
     ])
 
diff -up chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py.py2 chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
--- chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py.py2	2018-07-25 14:14:02.004886479 -0400
+++ chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py	2018-07-25 14:15:30.167896461 -0400
@@ -58,7 +58,7 @@ def _MinifyJS(input_js):
 
   with tempfile.NamedTemporaryFile() as _:
     args = [
-        'python',
+        'python2',
         rjsmin_path
     ]
     p = subprocess.Popen(args,
@@ -177,7 +177,7 @@ def _MinifyCSS(css_text):
       os.path.join(py_vulcanize_path, 'third_party', 'rcssmin', 'rcssmin.py'))
 
   with tempfile.NamedTemporaryFile() as _:
-    rcssmin_args = ['python', rcssmin_path]
+    rcssmin_args = ['python2', rcssmin_path]
     p = subprocess.Popen(rcssmin_args,
                          stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE,
