diff --git a/tools/gn/bootstrap/bootstrap.py b/tools/gn/bootstrap/bootstrap.py
index 3b2f4380854c5..53a03ae4d9f41 100755
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -115,6 +115,8 @@ def main(argv):
       os.path.join(GN_ROOT, 'build', 'gen.py'),
       '--no-last-commit-position',
       '--out-path=' + gn_build_dir,
+      '--no-static-libstdc++',
+      '--allow-warnings',
   ]
   if options.debug:
     cmd.append('--debug')
