diff --git a/device/base/synchronization/one_writer_seqlock.h b/device/base/synchronization/one_writer_seqlock.h
index 528b6683ca270..e37a16b81be89 100644
--- a/device/base/synchronization/one_writer_seqlock.h
+++ b/device/base/synchronization/one_writer_seqlock.h
@@ -6,6 +6,7 @@
 #define DEVICE_BASE_SYNCHRONIZATION_ONE_WRITER_SEQLOCK_H_
 
 #include <atomic>
+#include <cstdint>
 #include <type_traits>
 
 #include "base/atomicops.h"
