diff --git a/quiche/quic/core/crypto/quic_hkdf.h b/quiche/quic/core/crypto/quic_hkdf.h
index 6a300ed0..3e30f1cb 100644
--- a/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 #define QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 
+#include <cstdint>
 #include <vector>
 
 #include "absl/strings/string_view.h"
diff --git a/quiche/quic/core/quic_connection_id.h b/quiche/quic/core/quic_connection_id.h
index 49943495..b4e25fec 100644
--- a/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
+++ b/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h
@@ -5,6 +5,7 @@
 #ifndef QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 #define QUICHE_QUIC_CORE_QUIC_CONNECTION_ID_H_
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
