diff --git a/components/password_manager/core/browser/generation/password_generator.h b/components/password_manager/core/browser/generation/password_generator.h
index b0d621ce35fcb..aa674d47798f7 100644
--- a/components/password_manager/core/browser/generation/password_generator.h
+++ b/components/password_manager/core/browser/generation/password_generator.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 #define COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_GENERATION_PASSWORD_GENERATOR_H_
 
+#include <cstdint>
 #include <string>
 
 
