--- chromium-69.0.3497.81/tools/gn/bootstrap/bootstrap.py.orig	2018-09-05 03:10:45.000000000 +0800
+++ chromium-69.0.3497.81/tools/gn/bootstrap/bootstrap.py	2018-09-08 11:15:16.819121260 +0800
@@ -88,6 +88,7 @@
     gn_gen_args += ' is_debug=false'
   subprocess.check_call([
       gn_path, 'gen', out_dir,
+      '--script-executable=/usr/bin/python2',
       '--args=%s' % gn_gen_args, "--root=" + SRC_ROOT
   ])
 
diff -up chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py.py2 chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py
--- chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py.py2	2018-07-25 14:14:02.004886479 -0400
+++ chromium-67.0.3396.99/third_party/catapult/common/py_vulcanize/py_vulcanize/generate.py	2018-07-25 14:15:30.167896461 -0400
@@ -58,7 +58,7 @@ def _MinifyJS(input_js):
 
   with tempfile.NamedTemporaryFile() as _:
     args = [
-        'python',
+        'python2',
         rjsmin_path
     ]
     p = subprocess.Popen(args,
@@ -177,7 +177,7 @@ def _MinifyCSS(css_text):
       os.path.join(py_vulcanize_path, 'third_party', 'rcssmin', 'rcssmin.py'))
 
   with tempfile.NamedTemporaryFile() as _:
-    rcssmin_args = ['python', rcssmin_path]
+    rcssmin_args = ['python2', rcssmin_path]
     p = subprocess.Popen(rcssmin_args,
                          stdin=subprocess.PIPE,
                          stdout=subprocess.PIPE,
