diff --git a/third_party/maldoca/src/maldoca/ole/header.h b/third_party/maldoca/src/maldoca/ole/header.h
index 174f704cc37e6..c4d56653a3485 100644
--- a/third_party/maldoca/src/maldoca/ole/header.h
+++ b/third_party/maldoca/src/maldoca/ole/header.h
@@ -43,6 +43,8 @@
 #ifndef MALDOCA_OLE_HEADER_H_
 #define MALDOCA_OLE_HEADER_H_
 
+#include <cstdint>
+
 #include "absl/strings/string_view.h"
 
 namespace maldoca {
