--- a/build/config/compiler/BUILD.gn      2016-10-13 03:02:53.000000000 +0800
+++ b/build/config/compiler/BUILD.gn      2016-10-19 22:00:03.528903842 +0800
@@ -228,6 +228,14 @@
     defines += [ "CR_CLANG_REVISION=\"$clang_revision\"" ]
   }
 
+  if (!is_win && !is_clang) {
+    # GCC 6+ can optimize away pointer comparisons to null. This is
+    # problematic as V8 encodes Values through tagged pointers and comparisons
+    # with 0 are actually necessary in many cases. As a temporary Workaround
+    # we disable this optimization. See: https://crbug.com/v8/3782
+    cflags += [ "-fno-delete-null-pointer-checks" ]
+  }
+
   # Non-Mac Posix compiler flags setup.
   # -----------------------------------
   if (is_posix && !(is_mac || is_ios)) {
